### ISSUE TITLE
Detect and repair missing active stage for the internal _api config package

### ISSUE THREAD

**Issue by dnsmichi** [Icinga/icinga2#6959]
## Expected Behavior

Detect missing active-stages for the internal `_api` package.

## Current Behavior

Sometimes concurrent writes destroy the inner `active-stage` and `include.conf` files. Apparently the read from `active-stage` fails and as such, one part of `GetConfigDir()` is missing.

## Possible Solution

Fetch the last modified stage, and activate it again thus repairing the broken files.

```
michi@Michaels-MacBook-Pro ~/dev/icinga/icinga2 (bugfix/objects-no-stop *) $ git diff lib/remote/configobjectutility.cpp
diff --git a/lib/remote/configobjectutility.cpp b/lib/remote/configobjectutility.cpp
index 5575cba00..dee7155dd 100644
--- a/lib/remote/configobjectutility.cpp
+++ b/lib/remote/configobjectutility.cpp
@@ -32,8 +32,16 @@ using namespace icinga;

 String ConfigObjectUtility::GetConfigDir()
 {
-       return ConfigPackageUtility::GetPackageDir() + "/_api/" +
-               ConfigPackageUtility::GetActiveStage("_api");
+       String activeStage = ConfigPackageUtility::GetActiveStage("_api");
+
+       if (activeStage.IsEmpty()) {
+               Log(LogWarning, "ConfigObjectUtility")
+                       << "Could not fetch active stage for config package '_api'. Recreating and migrating.";
+
+
+       }
+
+       return ConfigPackageUtility::GetPackageDir() + "/_api/" + activeStage;
 }
```

## Steps to Reproduce (for bugs)

1. Create many downtimes/comments in parallel
2. Restart via /v1/actions/process-restart
3. Forcefully delete active-stage and see what Icinga does

## Your Environment

* Version used (`icinga2 --version`): 2.10.2 and pre 2.11
* Operating System and version: macOS mojave

**Comment by dnsmichi** [comment]
Actually it is not broken, the full analysis in in #7119. I'm closing this in favour of the mentioned issue.

**Comment by dnsmichi** [comment]
Will push a real fix for automated repairing if detected.
